syntax = "proto3";

message PersonIdMessage {
    int32 id = 1;
}

message PersonMessage {
    int32 id = 1;
    string first_name = 2;
    string last_name = 3;
    string company_name = 4;
    string registration_time = 5;
}

message PersonMessageList {
    repeated PersonMessage persons = 1;
}

message LocationIdMessage {
    int32 id = 1;
}

message LocationMessage {
    optional int32 id = 1;
    int32 person_id = 2;
    float longitude = 3;
    float latitude = 4;
    string creation_time = 5;
}

message LocationMessageList {
    repeated LocationMessage locations = 1;
}

message ConnectionMessage {
    PersonMessage person = 1;
    LocationMessage location = 2;
}

message ConnectionMessageList {
    repeated ConnectionMessage connections = 1;
}

message EmptyMessage {}


service PersonService {
    rpc GetPerson(PersonIdMessage) returns (PersonMessage);
    rpc GetPersons(EmptyMessage) returns (PersonMessageList);
    rpc CreatePerson(PersonMessage) returns (PersonMessage);
}

service ConnectionService {
    rpc GetConnections(PersonIdMessage) returns (ConnectionMessageList);
}
